packages/internal/test/test-callback.js

```diff
@@ -253,3 +264,52 @@ test('isCallback', t => {
     'missing bound args',
   );
 });
+
+test('makeAttenuator', async t => {
+  const makeAttenuator = cb.prepareAttenuator({
+    methodNames: ['m1', 'm2', 'm4'],
+  });
+  const original = Far('original', {
+    m1() {
+      return 'return original.m1';
+    },
+    m2() {
+      throw Error('unexpected original.m2');
+    },
+    m3() {
+      throw Error('unexpected original.m3');
+    },
+  });
+  const originalCbs = objectMap(original, (_, key) =>
+    cb.makeSyncMethodCallback(original, key),
+  );
+  t.throws(() => makeAttenuator(originalCbs), {
+    message: `Attenuator callbacks["m3"] not allowed by methodKeys`,
+  });
+
+  // Null out a method.
+  const at1 = makeAttenuator({ ...originalCbs, m3: null });
+  t.is(at1.m1(), 'return original.m1');
+  t.throws(() => at1.m2(), { message: `unexpected original.m2` });
+  t.throws(() => at1.m3(), { message: `at1.m3 is not a function` });
+  // @ts-expect-error deliberate: m4 is not a method
```

Comment:
`m4` is declared on the attenuator, I would have expected the types to derive from the definition of the attenuator and not from the instantiation. It's return type could be typed as `never`, but I don't think the method itself should be missing in the object's type.